File: Programming Basics/Java/Conditional Statements/TravelAgency.java
package com.company;

import java.util.Scanner;

public class TravelAgency {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String town = scanner.nextLine();
        String paketType = scanner.nextLine();
        String vip = scanner.nextLine();
        int days = Integer.parseInt(scanner.nextLine());
        boolean invalidInput = false;
        double price = 0.0;

        if (!town.equals("Bansko") && !town.equals("Borovets") &&
                !town.equals("Varna") && !town.equals("Burgas")) {
            System.out.println("Invalid input!");
            invalidInput = true;
        }

        if (!paketType.equals("noEquipment") && !paketType.equals("withEquipment") &&
                !paketType.equals("noBreakfast") && !paketType.equals("withBreakfast")) {
            System.out.println("Invalid input!");
            invalidInput = true;
        }

        if (days < 1) {
            System.out.println("Days must be positive number!");
            invalidInput = true;
        }

        if (days > 7) {
            days--;
        }

        if (!invalidInput) {
            if (town.equals("Bansko") || town.equals("Borovets")) {
                if (paketType.equals("noEquipment")) {
                    if (vip.equals("yes")) {
                        price = days * 80 * 0.95;
                    } else {
                        price = days * 80;
                    }
                } else if (paketType.equals("withEquipment")) {
                    if (vip.equals("yes")) {
                        price = days * 100 * 0.90;
                    } else {
                        price = days * 100;
                    }
                }
                System.out.printf("The price is %.2flv! Have a nice time!", price);
            } else if (town.equals("Varna") || town.equals("Burgas")) {
                if (paketType.equals("noBreakfast")) {
                    if (vip.equals("yes")) {
                        price = days * 100 * 0.93;
                    } else {
                        price = days * 100;
                    }
                } else if (paketType.equals("withBreakfast")) {
                    if (vip.equals("yes")) {
                        price = days * 130 * 0.88;
                    } else {
                        price = days * 130;
                    }
                }
                System.out.printf("The price is %.2flv! Have a nice time!", price);
            }

        }

    }
}

File: Programming Basics/JavaScript/Exam/Travel Agency.js
function solve(input) {
    let town = input.shift();
    let packetType = input.shift();
    let vipDiscount = input.shift();
    let days = Number(input.shift());
    let price = 0;
    let invalidInput = false;
    if (days < 1) {
        invalidInput = true;
        console.log("Days must be positive number!")
    } else {
        switch (town) {
            case "Bansko":
            case "Borovets":
                if (days > 7) {
                    days--
                }
                switch (packetType) {
                    case "noEquipment":
                        if (vipDiscount == "yes") {
                            price = 0.95 * (80 * days);
                        } else {
                            price = 80 * days;
                        }
                        ; break;
                    case "withEquipment":
                        if (vipDiscount == "yes") {
                            price = 0.90 * (100 * days);
                        } else {
                            price = 100 * days;
                        }
                        ; break;
                }; break;
            case "Varna":
            case "Burgas":
                if (days > 7) {
                    days--
                }
                switch (packetType) {
                    case "noBreakfast":
                        if (vipDiscount == "yes") {
                            price = 0.83 * (100 * days);
                        } else {
                            price = 100 * days;
                        }
                        ; break;
                    case "withBreakfast":
                        if (vipDiscount == "yes") {
                            price = 0.88 * (130 * days);
                        } else {
                            price = 130 * days;
                        }
                        ; break;
                    default:
                        invalidInput = true;
                        console.log("Invalid input!")
                }break;
            default:
                invalidInput = true;
                console.log("Invalid input!")
        }
    }
    if (invalidInput != true) {
        console.log(`The price is ${price.toFixed(2)}lv! Have a nice time!`);
    }
}
solve(["Gabrovo",
    "noBreakfast",
    "no",
    3])
